examples/image_classification/ios/ImageClassifier/ViewContoller/ViewController.swift

```diff
@@ -0,0 +1,400 @@
+// Copyright 2023 The MediaPipe Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import MediaPipeTasksVision
+import UniformTypeIdentifiers
+import AVKit
+
+class ViewController: UIViewController {
+
+  // MARK: Storyboards Connections
+  @IBOutlet weak var previewView: PreviewView!
+  @IBOutlet weak var addImageButton: UIButton!
+  @IBOutlet weak var cameraUnavailableLabel: UILabel!
+  @IBOutlet weak var imageEmptyLabel: UILabel!
+  @IBOutlet weak var resumeButton: UIButton!
+  @IBOutlet weak var runningModelTabbar: UITabBar!
+  @IBOutlet weak var cameraTabbarItem: UITabBarItem!
+  @IBOutlet weak var photoTabbarItem: UITabBarItem!
+  @IBOutlet weak var bottomSheetViewBottomSpace: NSLayoutConstraint!
+  @IBOutlet weak var bottomViewHeightConstraint: NSLayoutConstraint!
+
+  // MARK: Constants
+  private let delayBetweenInferencesMs = 1000.0
+  private let inferenceBottomHeight = 220.0
+  private let expandButtonHeight = 41.0
+  private let playerViewController = AVPlayerViewController()
+  private let videoOutput = AVPlayerItemVideoOutput(pixelBufferAttributes: [String(kCVPixelBufferPixelFormatTypeKey): NSNumber(value: kCVPixelFormatType_32BGRA)])
+
+  // MARK: Instance Variables
+  private var videoDetectTimer: Timer?
+  private var previousInferenceTimeMs = Date.distantPast.timeIntervalSince1970 * 1000
+  private var maxResults = DefaultConstants.maxResults {
+    didSet {
+      guard let inferenceVC = inferenceViewController else { return }
+      bottomViewHeightConstraint.constant = inferenceVC.collapsedHeight + inferenceBottomHeight
+      view.layoutSubviews()
+    }
+  }
+  private var liveStreamDelegate: ImageClassifierLiveStreamDelegate?
```

Comment:
We want to do everything related to the ImageClassifier in a single class, which is ImageClassifierHelper. You should make ImageClassifierHelper to follow ImageClassifierLiveStreamDelegate protocol, and then return the detection result with the inference time in a ResultBundle object to this ViewController class. See the Android code for `ClassifierListener` reference.
https://github.com/googlesamples/mediapipe/blob/39ce84ae02628d7891d8a6656c7e8de860d31df8/examples/image_classification/android/app/src/main/java/com/google/mediapipe/examples/imageclassification/ImageClassifierHelper.kt#L44C9-L44C32